### ISSUE TITLE
Enhanced release alerts

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/balloons-and-static-electricity#337]
To track progress in  implementing the updated a11y designs listed in

https://docs.google.com/document/d/1axeAHQZvx6lOpcSywRYvAFvs3xJrMA5vcXkA-AJTnhs/edit#

**Comment by jessegreenberg** [comment]
Mostly done in the above commit. This sounds very cool! Usually we don't here that the balloon is moving through the center of the play area because it isn't in that landmark for long enough, even with only one charge.

Current logic is every 3 seconds or so, we say "Moving {{direction}}." or "Moving {{direction}}. {{location}}." if balloon in a landmark. So we hear the movement progress more clearly, I would like to try announcing this alert every time landmark changes during independent movement as well.

**Comment by jessegreenberg** [comment]
I tried the change, but didn't like it very much because the alert happened too frequently when it entered landmarks around the sweater.

**Comment by jessegreenberg** [comment]
@terracoda I think everything from "Release Movement 4 Complex Path Descriptions" doc is implemented, could you please review?

**Comment by jessegreenberg** [comment]
On slack @terracoda said

> I also wonder =ing if we should just go with location. I would be more than willing to try that.

Does this mean only use landmarks and not have a delay?

**Comment by jessegreenberg** [comment]
Over slack @terracoda said:

>It would be wonderful to have regular intervals of "Moving {{direction}}" and then always hear {{Moving {{direction}}. {{Landmark location}}. It might be necessary to restart the timing count after a landmark, but we can see about that.

**Comment by jessegreenberg** [comment]
Done in the above commit @terracoda could you please review at your convenience?

**Comment by terracoda** [comment]
Thanks @jessegreenberg, since you are so speedy, I will review now.

**Comment by jessegreenberg** [comment]
Balloon wasn't announcing when it was on the sweater in independent movement, above commit fixes that.

**Comment by terracoda** [comment]
@jessegreenberg, on VO it sounds very good when velocity is slow (i.e., when alerts do not get interrupted).  Let me if there is a velocity threshold implemented and I will listen more in the morning.

**Comment by terracoda** [comment]
Awesome work, btw.

**Comment by jessegreenberg** [comment]
Thanks @terracoda!

> Let me if there is a velocity threshold implemented

I don't  see this listed in the design document, is there an example of this somewhere?

**Comment by terracoda** [comment]
@jessegreenberg, I just wanted to verify that you were just using the timer approach (approximately every 3 seconds) and nothing else to determine when alerts are announced/released.

**Comment by jessegreenberg** [comment]
Sorry that is not how I interpreted this:
>  It would be wonderful to have regular intervals of "Moving {{direction}}" and then always hear {{Moving {{direction}}. {{Landmark location}}. It might be necessary to restart the timing count after a landmark, but we can see about that.

Right now it works like this,  so
- Every ~3 seconds, we hear "Moving {{direction}}."
- We always hear "{{Moving {{direction}}. {{Landmark location}}." when the balloon enters a landmark on its own. This resets the timer.

**Comment by terracoda** [comment]
Thanks @jessegreenberg, that description is very clear.

**Comment by terracoda** [comment]
@jessegreenberg, would I be able to try a version that does not include the timer reset.

**Comment by terracoda** [comment]
@jessegreenberg, we definitely need a velocity threshold for the Release movement 4 (RM4 pattern) to be triggered. RM4 pattern is only meant for velocities slowly, very slowly, extremely slowly. 

There is not time for continued movement phrases and changes in directions when the balloon is moving quickly or very quickly.  Diagonally movement phrase captures some interesting release paths.

**Comment by jessegreenberg** [comment]
Understood, thanks @terracoda.

**Comment by terracoda** [comment]
@jessegreenberg, I think there was a velocity threshold when we were using "Still moving". 

Is limiting the RM4 pattern to velocities that start out as "slowly", "very slowly", "extremely slowly" workable?

**Comment by jessegreenberg** [comment]
This change was made in the above commit, so now the behavior is:

- If balloon moves slowly initially on release:
  - Every ~3 seconds, we hear "Moving {{direction}}."
  - We always hear "{{Moving {{direction}}. {{Landmark location}}." when the balloon enters a landmark on its own. This resets the timer.

@terracoda could you please review?

**Comment by jessegreenberg** [comment]
I found it wasn't working very well in master, above commit fixes a bug with this.

**Comment by jessegreenberg** [comment]
IMO it still feels like the RM4 alerts happen too often. Maybe they should only be heard if released "extremely slowly"? Or maybe we should change the ranges for when the balloon is considered to be moving "slowly"? Just thoughts for review @terracoda, let me know what you think.

**Comment by terracoda** [comment]
@jessegreenberg, that was my original idea..."only for extremely slowly."

Let me have another listen. During my last listen, I also thought it was too chatty, and in VO that means a lot of interruption.

**Comment by terracoda** [comment]
The Release Movement 4 pattern works really well when one balloon has very little charge:
<img width="982" alt="base-rm4_alerts-sound-good-here" src="https://user-images.githubusercontent.com/1902116/34111715-1e12ade2-e3e5-11e7-8a5e-98b0d320e1a3.png">

**Comment by terracoda** [comment]
And Release Movement 4 pattern works well with a few more charges as well, but there is no final "Release ending phrase" when the balloon either is released or moves independently and movement stops against the right edge when there is no wall present.

See the difference in the examples:

**Release ending at wall contains Release ending phrase**
<img width="974" alt="base-release-ending-at-wall" src="https://user-images.githubusercontent.com/1902116/34114398-4412f490-e3ed-11e7-83a2-49cb9ee84f91.png">

**Indy balloon movement ending at wall contains the Release ending phrase**
<img width="983" alt="base-indy-balloon-movement-ends-at-wall-with-release-ending" src="https://user-images.githubusercontent.com/1902116/34115078-1b7376de-e3ef-11e7-96a2-000aa5b270e2.png">

**Release ending a right edge DOES NOT contain the Release ending phrase**
<img width="973" alt="base-release-ending-at-right-edge-no-ending-phrase" src="https://user-images.githubusercontent.com/1902116/34114432-58a926d6-e3ed-11e7-80a7-acbd2bc951d3.png">

**Indy balloon movement ending at right edge DOES NOT contain the Release ending phrase.** I guess in this case, it would be a "movement ending phrase", but it would be the same as the "Release ending phrase".
<img width="973" alt="base-indy-movent-ending-at-right-edge" src="https://user-images.githubusercontent.com/1902116/34115572-9c8418f4-e3f0-11e7-8e96-b6172c0c7c0e.png">

**Comment by jessegreenberg** [comment]
Sounds good @terracoda, changes for https://github.com/phetsims/balloons-and-static-electricity/issues/337#issuecomment-352474399 made in the above commit. Any time velocity stops (whether or not balloon hits wall), we hear [[String: Balloon name]], [[String: Location]].

**Comment by terracoda** [comment]
Seems to work well in dev version 9. Closing issue.
